pkg/karmadactl/cmdinit/utils/util_test.go

```diff
@@ -1,26 +1,63 @@
 package utils
 
 import (
+	_ "embed"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
 	"os"
+	"path/filepath"
 	"testing"
 )
 
+// This file is created by:
+// mkdir -p testdata; cd testdata; echo test > test.txt; tar czf test.tar.gz test.txt; rm -f test.txt
+//go:embed testdata/test.tar.gz
+var testTarData []byte
+
 const (
-	filePath = "./crds.tar.gz"
-	folder   = "./crds"
+	testTar  = "test.tar.gz"
```

Comment:
I see the tar file is quite simple, can we just create it in the start of testing code and delete it after finish?

Reply:
done